quinn-proto/src/token.rs

```diff
@@ -75,8 +64,13 @@ impl RetryToken {
     }
 }
 
-fn encode_addr(buf: &mut Vec<u8>, address: &SocketAddr) {
-    match address.ip() {
+fn encode_socket_addr(buf: &mut Vec<u8>, address: &SocketAddr) {
```

Comment:
Nots: I'd prefer `encode_addr()` and `encode_ip()` over `encode_socket_addr()` and `encode_ip_addr()`.

Also, please sort by type, then direction instead of the other way around (so `decode_addr()` before `encode_ip()`).